model/common/src/icon4py/model/common/states/factory.py

```diff
@@ -213,6 +256,130 @@ def func(self) -> Callable:
         return lambda: self.fields
 
 
+class FieldOperatorProvider(FieldProvider):
+    """Provider that calls a GT4Py Fieldoperator.
+
+    # TODO (@halungge) for now to be used only on FieldView Embedded GT4Py backend.
+    - restrictions:
+         - (if only called on FieldView-Embedded, this is not a necessary restriction)
+            calls field operators without domain args, so it can only be used for full field computations
+    - plus:
+        - can write sparse/local fields
+    """
+
+    def __init__(
+        self,
+        func: gtx_decorator.FieldOperator,
+        domain: tuple[gtx.Dimension, ...],
+        fields: dict[str, str],  # keyword arg to (field_operator, field_name)
+        deps: dict[str, str],  # keyword arg to (field_operator, field_name) need: src
+        params: Optional[
+            dict[str, state_utils.ScalarType]
+        ] = None,  # keyword arg to (field_operator, field_name)
+    ):
+        self._func = func
+        self._dims = domain
+        self._dependencies = deps
+        self._output = fields
+        self._params = params if params is not None else {}
```

Comment:
```suggestion
        self._params = {} if params is None else params
```
just as a personal preference